pallets/parachain-staking/src/lib.rs

```diff
@@ -797,10 +807,10 @@ pub mod pallet {
                 candidate_count >= old_count,
                 Error::<T>::TooLowCandidateCountWeightHintJoinCandidates
             );
-            ensure!(
-                candidates.insert(Bond { owner: acc.clone(), amount: bond }),
-                Error::<T>::CandidateExists
-            );
+            let maybe_inserted_candidate = candidates
+                .try_insert(Bond { owner: acc.clone(), amount: bond })
+                .map_err(|_| Error::<T>::CandidateLimitReached)?;
+            ensure!(maybe_inserted_candidate, Error::<T>::CandidateExists);
```

Comment:
Is it possible to get a CanidateExists error if we have mapped it with CandidateLimitReached?
Would a match here be better to handle different error cases and drop the `ensure!` check?